theano/tensor/signal/downsample.py

```diff
@@ -333,6 +333,19 @@ def grad(self, inp, grads):
                                     st=self.st, padding=self.padding,
                                     mode=self.mode)(
                                         x, gz)]
+
+    def R_op(self, inputs, eval_points):
+        # R_op can receive None as eval_points.
+        # That mean there is no diferientiable path through that input
+        # If this imply that you cannot compute some outputs,
+        # return None for those.
+        if eval_points[0] is None:
+            return [None]
+        rop = DownsampleFactorMaxRop(self.ds,
```

Comment:
Here, until padding and strides are supported raise NotImplementedError when padding isn't 0 and strides isn't the same as ds.

Do you need those feature?